kernels/optimized/cpu/op_linear.cpp

```diff
@@ -56,6 +50,25 @@ Tensor& opt_linear_out(
         size_t k = in.sizes()[in.dim() - 1];
         size_t m = mat2.size(0);
 
+        // If bias is provided, verify its shape and pre-fill the output tensor.
+        if (bias.has_value()) {
+          auto bias_value = bias.value();
+          // Check that bias is 1D and its size matches m.
+          ET_KERNEL_CHECK(ctx, bias_value.dim() == 1 && bias_value.size(0) == m, InvalidArgument, out);
```

Comment:
it would help users debug if this was an ET_KERNEL_CHECK_MSG that printed the values of bias_value.dim(), bias_value.size(0), and m when it failed